### ISSUE TITLE
Multiple errors in newest update

### ISSUE THREAD

**Issue by Rusturion** [Vectorial1024/Desynchronized#10]
For reference, I no longer had Runtime GC installed. Reinstalling Runtime GC stopped this error from occurring.

Every 6 hours the log gets spammed with a variation of the below message for _every pawn_.

Desynchronized has disappeared entirely from Steam, so I feel like you may be aware of the issue.
Please let me know if you'd like further assistance in testing whilst you track down/resolve this issue.

https://gist.github.com/HugsLibRecordKeeper/6ce36b6218f775b23b739ad0d776ba33

[V1024-DESYNC]  Requesting GC status for pawn Donny; game gave reason of 
Verse.Log:Error(String, Boolean)
Desynchronized.DesynchronizedMain:LogError(String, Boolean)
Desynchronized.Patches.PostFix_WorldPawnGC_PawnImportanceReason:PostFix(Pawn, String&)
RimWorld.Planet.WorldPawnGC:GetCriticalPawnReason_Patch1(Object, Pawn)
RimWorld.Planet.<AccumulatePawnGCData>c__Iterator0:MoveNext()
RimWorld.Planet.<PawnGCPass>c__Iterator1:MoveNext()
RimWorld.Planet.WorldPawnGC:WorldPawnGCTick()
RimWorld.Planet.WorldPawns:WorldPawnsTick()
RimWorld.Planet.World:WorldTick()
Verse.TickManager:DoSingleTick()
Verse.TickManager:TickManagerUpdate()
Verse.Game:UpdatePlay()
Verse.Root_Play:Update()

**Comment by Vectorial1024** [comment]
Ah shit. No wonder why it seemed suspiciously quiet these two days after the update. Will respond in a few hours, since I'm not at home right now.

**Comment by Rusturion** [comment]
Just received 4 of these:
```
Faction  has null relation with . Returning dummy relation.
Verse.Log:Error(String, Boolean)
RimWorld.Faction:RelationWith(Faction, Boolean)
RimWorld.Faction:RelationKindWith(Faction)
Desynchronized.GeneralExtensionHelper:AlliedTo(Faction, Faction)
Desynchronized.TNDBS.Utilities.NewsImportanceExtensionHelper:GetSocialProximityScoreForOther(Pawn, Pawn)
Desynchronized.TNDBS.TaleNewsPawnDied:CalculateNewsImportanceForPawn(Pawn, TaleNewsReference)
Desynchronized.TNDBS.NewsSpreadUtility:SelectNewsRandomly(Pawn, Pawn, TaleNewsReference&)
Desynchronized.TNDBS.NewsSpreadUtility:DetermineTaleNewsToTransmit(Pawn, Pawn, SpreadMode)
Desynchronized.TNDBS.NewsSpreadUtility:SpreadNews(Pawn, Pawn, SpreadMode)
Desynchronized.Patches.PostFix_InteractionWorker:PostFix(InteractionWorker, Pawn, Pawn)
RimWorld.InteractionWorker:Interacted_Patch1(Object, Pawn, Pawn, List`1, String&, String&, LetterDef&)
RimWorld.Pawn_InteractionsTracker:TryInteractWith_Patch1(Object, Pawn, InteractionDef)
Hospitality.Harmony.TryInteractRandomly:Replacement(Pawn_InteractionsTracker, Boolean&)
RimWorld.Pawn_InteractionsTracker:TryInteractRandomly_Patch1(Object)
RimWorld.Pawn_InteractionsTracker:InteractionsTrackerTick()
Verse.Pawn:Tick_Patch1(Object)
Verse.TickList:Tick_Patch2(Object)
Verse.TickManager:DoSingleTick()
Verse.TickManager:TickManagerUpdate()
Verse.Game:UpdatePlay()
Verse.Root_Play:Update()
```
And then one of these:
```
Exception ticking Fiddler (at (92, 0, 172)): System.ArgumentOutOfRangeException: Argument is out of range.
Parameter name: index
at System.Collections.Generic.List`1<Desynchronized.TNDBS.TaleNewsReference>.get_Item (int) <0x00083>
at Desynchronized.TNDBS.NewsSpreadUtility.SelectNewsRandomly (Verse.Pawn,Verse.Pawn,Desynchronized.TNDBS.TaleNewsReference&) <0x00365>
at Desynchronized.TNDBS.NewsSpreadUtility.DetermineTaleNewsToTransmit (Verse.Pawn,Verse.Pawn,Desynchronized.TNDBS.NewsSpreadUtility/SpreadMode) <0x0006b>
at Desynchronized.TNDBS.NewsSpreadUtility.SpreadNews (Verse.Pawn,Verse.Pawn,Desynchronized.TNDBS.NewsSpreadUtility/SpreadMode) <0x00025>
at Desynchronized.Patches.PostFix_InteractionWorker.PostFix (RimWorld.InteractionWorker,Verse.Pawn,Verse.Pawn) <0x000cc>
at (wrapper dynamic-method) RimWorld.InteractionWorker.Interacted_Patch1 (object,Verse.Pawn,Verse.Pawn,System.Collections.Generic.List`1<Verse.RulePackDef>,string&,string&,Verse.LetterDef&) <0x00051>
at (wrapper dynamic-method) RimWorld.Pawn_InteractionsTracker.TryInteractWith_Patch1 (object,Verse.Pawn,RimWorld.InteractionDef) <0x005b2>
at Hospitality.Harmony.Pawn_InteractionsTracker_Patch/TryInteractRandomly.Replacement (RimWorld.Pawn_InteractionsTracker,bool&) <0x00433>
at (wrapper dynamic-method) RimWorld.Pawn_InteractionsTracker.TryInteractRandomly_Patch1 (object) <0x0003d>
at RimWorld.Pawn_InteractionsTracker.InteractionsTrackerTick () <0x0016a>
at (wrapper dynamic-method) Verse.Pawn.Tick_Patch1 (object) <0x003d7>
at (wrapper dynamic-method) Verse.TickList.Tick_Patch2 (object) <0x00497>
```
The weirdest part is, this occurred on day 2, and no pawn has died yet.
I'll try not to get too attached to this colony haha
[Full log](https://gist.github.com/HugsLibRecordKeeper/5e53d13beecc073711214fe67ffb855e)

**Comment by Vectorial1024** [comment]
Apparently Valve and Steam sneaked in a new EULA recently, which I did not notice when I updated the mod on Steam. That should be fixed now.

The errors are harmless, but then again, those errors are left over from development. (Facepalm)

My mod does not directly interact with Runtime GC. My guess is that Runtime GC is interfacing with the `WorldPawns` class directly, skipping the whatever method that I am trying to patch, so the "error" within my patch method disappears. This is a problematic situation for me.

Gotta explain what I'm trying to do for a bit. Previously, pawns referenced by this mod are not treated specially, and will be discarded by the game when they appear to be "dereferenced" (not in combat logs, not in graves, no where to be found, etc.). Now, by checking every pawn to see if they are referenced by any Tale-News in this mod, I can tell the game "hey I need this" so the game will not discard them.

Faction relation is one of the several factors when calculating the news importance. But then, I thought, "hey, maybe the game can handle these boundary cases of null-null faction relationships so let's ignore those cases". Apparently, the game throws errors for that. Gonna fix that.

As for the argument-out-of-range exception, I guess I forgot to put in the "default tale news" when there are no tale-news to be selected.

I swear to god, this has to be the worst update I've ever made... so many leftover food lying around getting spoiled...

**Comment by Rusturion** [comment]
In the [first comment](https://github.com/Vectorial1024/Desynchronized/issues/10#issue-442490465) I did not have RuntimeGC installed. The reference to GC must be the vanilla Game Cleaner, which Runtime GC is a replacement/upgrade of.

The strange part is that resubscribing to RuntimeGC actually prevents that first issue.

I've renamed the issue, as the title was misleading due to my misunderstanding, and originally assuming GC referred to the mod RuntimeGC.
